src/pages/ListingPage/index.js

```diff
@@ -11,9 +11,14 @@ const ListingPage = ({ firebase }) => {
   const [offers, setOffers] = useState([]);
 
   useEffect(() => {
-    firebase.database
-      .ref("offers")
-      .once("value", snapshot => setOffers(Object.values(snapshot.val())));
+    firebase.firestore
+      .collection("offer")
+      .get()
+      .then(querySnapshot => {
+        const docs = [];
+        querySnapshot.forEach(doc => docs.push({ id: doc.id, ...doc.data() }));
```

Comment:
Why not use `map`?

Reply:
QuerySnapshot is not a normal collection, it is QuerySnapshot class and this class does not contain a `map` method